packages/@angular/cli/commands/completion.ts

```diff
@@ -117,14 +117,15 @@ const CompletionCommand = Command.extend({
         });
       }
 
-      let opts = '';
+      let opts: string[] = [];
       if (command.blueprints && command.blueprints[0]) {
-        opts += extractBlueprints(command.blueprints);
+        opts = opts.concat(extractBlueprints(command.blueprints));
       }
 
       if (command.availableOptions && command.availableOptions[0]) {
```

Comment:
I am not sure how to test this, and - IMO - is a bit out of scope for this simple fix.

A test of the whole completion feature would require to run a bash, source the ng completion output, then run complete on a string.

Here is a simple helper shell script to test the completion test_completion.sh
```
#!/bin/bash
ARGS=$@
# load bash-completion helper functions
source /usr/share/bash-completion/bash_completion
# array of words in command line
COMP_WORDS=($ARGS)
# index of the word containing cursor position
COMP_CWORD=$#
# command line
COMP_LINE="$ARGS"

# index of cursor position
COMP_POINT=${#ARGS}
source <(./bin/ng completion 2>/dev/null)

_xfunc ng _ng_completion

# print completions to stdout
printf '%s\n' "${COMPREPLY[@]}"
```

Some sample output:
```
$ ./test_completion.sh ng generate
class
component
directive
enum
guard
interface
module
pipe
service--dry-run
--verbose
-d
-v
```

After the commit is applied:
```
$ ./test_completion.sh ng generate
--dry-run
--verbose
-d
-v
class
component
directive
enum
guard
interface
module
pipe
service
```

An option would be to run this script for every command, and then check, for each line of the output, if the string is either a blueprint or a "--option" or a "-alias"

But I am not sure how to do this - this might be a separate bug/issue to add tests for completion...